packages/client/src/extension.mts

```diff
@@ -193,9 +195,47 @@ async function _activate(options: ActivateOptions): Promise<ExtensionApi> {
         client.onBlockFile(notifyUserOfBlockedFile),
     );
 
+    // If auto-correct is enabled, add the handler for it.
+    const config = vscode.workspace.getConfiguration();
+    if (config.get('cSpell.autoCorrect', false)) {
```

Comment:
Adding the check here means that it is necessary to restart the extension before the autocorrect setting takes effect. This can be very confusing to users.

It would better to always register the `onDidChangeTextDocument` handler and check to see if action is needed in `handleOnDidChangeTextDocument`